src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.ts

```diff
@@ -159,22 +162,24 @@ export class InstructorHomePageComponent extends InstructorSessionBasePageCompon
    * Loads the feedback session in the course.
    */
   loadFeedbackSessions(model: CourseTabModel): void {
-    this.httpRequestService.get('/sessions', {
-      courseid: model.course.courseId,
-    }).subscribe((response: FeedbackSessions) => {
-      response.feedbackSessions.forEach((feedbackSession: FeedbackSession) => {
-        const m: SessionsTableRowModel = {
-          feedbackSession,
-          responseRate: '',
-          isLoadingResponseRate: false,
-          instructorPrivilege: defaultInstructorPrivilege,
-        };
-        model.sessionsTableRowModels.push(m);
-        this.updateInstructorPrivilege(m);
-      });
-
-      model.isTabExpanded = true;
-    }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
+    if (!model.hasPopulated) {
+      this.httpRequestService.get('/sessions', {
+        courseid: model.course.courseId,
+      }).subscribe((response: FeedbackSessions) => {
+        response.feedbackSessions.forEach((feedbackSession: FeedbackSession) => {
+          const m: SessionsTableRowModel = {
+            feedbackSession,
+            responseRate: '',
+            isLoadingResponseRate: false,
+            instructorPrivilege: defaultInstructorPrivilege,
+          };
+          model.sessionsTableRowModels.push(m);
+          this.updateInstructorPrivilege(m);
+        });
+      }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
+      model.hasPopulated = true;
```

Comment:
Let's now take this chance to improve the UX.

You can see here that regardless of the API response, the row will be considered "populated". In case of API failure, it is clearly not the case. You may want to add some kind of error message to user and allow them to reload the feedback sessions in such a case.